### PR TITLE
Add documentation about internals

### PR DESCRIPTION
**Pull request by leandro-lucarella-sociomantic**

### CODE REVIEW

**Comment by leandro-lucarella-sociomantic** [comment]
BTW, asking for review mostly to catch major errors, not expecting you to get familiar with the internals for real or do an in-depth check of the correctness of the info.

File: INTERNALS.md [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+The main component of this is the [`beaver`](bin/beaver) binary. This script
+will only forward arguments to sub-commands, which are implemented as
+independent binaries, making sure the binaries directory is in the path,
+providing help and listing facilities for sub-commands, etc.
+
+Sub-commands can have two levels, which are handled by `beaver` and
+auto-discovered based on the directory structure. Let's say the user runs
+`beaver arg1 arg2 arg3`, then if an executable named `beaver-arg1` along where
+`beaver` leaves, it will run `beaver-arg1 arg2 arg3`. If it doesn't, but there
```
Comment (ben-palmer-sociomantic) [inline INTERNALS.md]: I think `leaves` here should be `lives`

**Review by ben-palmer-sociomantic — APPROVED** [review]
Only one minor comment from me.

File: INTERNALS.md (hunk 1 above)
Comment (leandro-lucarella-sociomantic) [reply INTERNALS.md]: Your right :troll:

**Pull request merged**